metron-platform/metron-common/src/main/java/org/apache/metron/common/configuration/ConfigurationsUtils.java

```diff
@@ -572,15 +590,22 @@ public static void applyConfigPatchToZookeeper(ConfigurationType configurationTy
    * @param patchData a JSON patch in the format specified by RFC 6902
    * @param client access to zookeeeper
    */
-  public static void applyConfigPatchToZookeeper(ConfigurationType configurationType,
-      Optional<String> configName,
-      byte[] patchData, CuratorFramework client) throws Exception {
+  public static void applyConfigPatchToZookeeper(
+          ConfigurationType configurationType,
+          Optional<String> configName,
+          byte[] patchData, CuratorFramework client) throws Exception {
+
     byte[] configData = readConfigBytesFromZookeeper(configurationType, configName, client);
     JsonNode source = JSONUtils.INSTANCE.readTree(configData);
     JsonNode patch = JSONUtils.INSTANCE.readTree(patchData);
     JsonNode patchedConfig = JSONUtils.INSTANCE.applyPatch(patch, source);
-    writeConfigToZookeeper(configurationType, configName,
-        JSONUtils.INSTANCE.toJSONPretty(patchedConfig), client);
+    byte[] prettyPatchedConfig = JSONUtils.INSTANCE.toJSONPretty(patchedConfig);
+
+    // ensure the patch produces a valid result; otherwise exception thrown during deserialization
```

Comment:
Can we reword this?  It needs to be clear that some things cannot be deserialized if they have an invalid configuration.

Reply:
I'm not sure exactly what you're looking for.  Can you suggest some specific text?

Reply:
```java
//Some configurations have side effects to deserialization, and may throw exceptions from 
//validation that may happen during the deserialization.  This does not meant that the //deserialization has failed due to invalid serialization.
```
?

Reply:
Never mind Nick, I didn't read this correctly, and I have the "serialization mixed with logical validation" stuff upfront in my head because of my work over in https://github.com/apache/metron/pull/856.

Sorry

Reply:
Heh, well our meta-PR was to address a code change that resulted in duplication, not just a style preference. ;)